app/services/courses/copy_to_provider_service.rb

```diff
@@ -1,29 +1,31 @@
 module Courses
   class CopyToProviderService
-    def initialize(course:)
-      @course = course
+    def initialize(sites_copy_to_course:, enrichments_copy_to_course:)
+      @sites_copy_to_course = sites_copy_to_course
+      @enrichments_copy_to_course = enrichments_copy_to_course
     end
 
-    def execute(new_provider)
-      new_course = new_provider.courses.find_by(course_code: @course.course_code)
+    def execute(course:, new_provider:)
+      new_course = new_provider.courses.find_by(course_code: course.course_code)
 
       return nil if new_course.present?
```

Comment:
These two lines could be refactored into a method that would read a bit nicer (and remove the need to reassign `new_course` further down)

something like:

```ruby
return nil if course_code_already_exists_on_provider?
```